### PR TITLE
Adding headers to HTTPError

### PR DESCRIPTION
**Pull request by teaglebuilt**
**In Response to Issue #275 **
[ Issue #275 ](https://github.com/bocadilloproject/bocadillo/issues/275)

I wanted to see if this is what you had in mind? I also read the link regarding the issue about adding to HTTPException and abondoning HTTPError?
[#507](https://github.com/encode/starlette/pull/507)

If you noticed I just added the headers to HTTPError and the handlers. I wanted to see which direction you are referring to?

### CODE REVIEW

File: bocadillo/error_handlers.py [hunk 1]
```diff
@@ -46,6 +48,7 @@
     ```
     """
     res.status_code = exc.status_code
+    res.headers = exc.headers
```
Comment (florimondmanca) [inline bocadillo/error_handlers.py]: I think some of the tests failing in CI come from this line. We shouldn’t set response headers if `exc.headers is None`. :)

**Comment by florimondmanca** [comment]
Yup, I think this is the way I initially thought about this. :)

File: bocadillo/error_handlers.py (hunk 1 above)
Comment (teaglebuilt) [reply bocadillo/error_handlers.py]: oh duh its optional, ill change that. well what about adding it to __slots__?

Comment (teaglebuilt) [reply bocadillo/error_handlers.py]: __slots__ = (status, detail, headers ) -> ?

Comment (florimondmanca) [reply bocadillo/error_handlers.py]: We can certainly add it, yes.

**Comment by florimondmanca** [comment]
@teaglebuilt Bocadillo has now entered unmaintained state, as per #334. I'm closing this PR as a result. Thanks for your interest. :)

**Pull request closed without merging**